modules/administration-guide/partials/proc_migration-to-devworkspace-engine.adoc

```diff
@@ -0,0 +1,209 @@
+IDENTITY_PROVIDER_SECRET=\$({orch-cli} get checluster/{prod-checluster} -n {prod-namespace} -o jsonpath="{.spec.auth.identityProviderSecret}")
+IDENTITY_PROVIDER_PASSWORD=\$(if [ -z "\$IDENTITY_PROVIDER_SECRET" ] || [ \$IDENTITY_PROVIDER_SECRET = "null" ]; then {orch-cli} get checluster/{prod-checluster}  -n {prod-namespace} -o jsonpath="{.spec.auth.identityProviderPassword}"; else {orch-cli} get secret \$IDENTITY_PROVIDER_SECRET -n {prod-namespace} -o jsonpath="{.data.password}" | base64 -d; fi)
+IDENTITY_PROVIDER_USERNAME=\$(if [ -z "\$IDENTITY_PROVIDER_SECRET" ] || [ \$IDENTITY_PROVIDER_SECRET = "null" ]; then {orch-cli} get checluster/{prod-checluster}  -n {prod-namespace} -o jsonpath="{.spec.auth.IdentityProviderAdminUserName}"; else {orch-cli} get secret \$IDENTITY_PROVIDER_SECRET -n {prod-namespace} -o jsonpath="{.data.user}" | base64 -d; fi)
+IDENTITY_PROVIDER_REALM=\$({orch-cli} get checluster {prod-checluster} -n {prod-namespace} -o jsonpath="{.spec.auth.identityProviderRealm}")
+
+refreshToken() {
+    IDENTITY_PROVIDER_TOKEN=\$(curl -ks \
+    -d "client_id=admin-cli" \
+    -d "username=\${IDENTITY_PROVIDER_USERNAME}" \
+    -d "password=\${IDENTITY_PROVIDER_PASSWORD}" \
+    -d "grant_type=password" \
+    "\${IDENTITY_PROVIDER_URL}/realms/master/protocol/openid-connect/token" | jq -r ".access_token")
+}
+
+run() {
+  rm -f \${ALL_USERS}
+  refreshToken
+  USER_IDS=(\$(curl -ks  -H "Authorization: bearer \${IDENTITY_PROVIDER_TOKEN}" "\${IDENTITY_PROVIDER_URL}/\${IDENTITY_PROVIDER_USERNAME}/realms/\${IDENTITY_PROVIDER_REALM}/users" | jq ".[] | .id" | tr "\r\n" " "))
+
+  for USER_ID in "\${USER_IDS[@]}"; do
+      refreshToken
+
+      USER_ID=\$(echo "\${USER_ID}" | tr -d "\"")
+      FEDERATED_IDENTITY=\$(curl -ks -H "Authorization: bearer \${IDENTITY_PROVIDER_TOKEN}" "\${IDENTITY_PROVIDER_URL}/\${IDENTITY_PROVIDER_USERNAME}/realms/\${IDENTITY_PROVIDER_REALM}/users/\${USER_ID}/federated-identity")
+      IDENTITY_PROVIDER=\$(echo "\${FEDERATED_IDENTITY}" | jq -r ".[] | select(.identityProvider == \"openshift-v4\")")
+      if [ -n "\${IDENTITY_PROVIDER}" ]; then
+        OPENSHIFT_USER_ID=\$(echo "\${IDENTITY_PROVIDER}" | jq ".userId" | tr -d "\"")
+        echo "[INFO] Find {prod-short} user: \${USER_ID} and corresponding OpenShift user: \${OPENSHIFT_USER_ID}"
+        echo "\${USER_ID} \${OPENSHIFT_USER_ID}" >> \${ALL_USERS}
+      fi
+  done
+}
+
+run
+echo "[INFO] Done."
+EOF
+
+bash get-all-{prod-id}-users.sh
+----
+
+include::partial$snip_scaling-down-keycloak.adoc[]
+
+include::partial$snip_finding-che-database-name.adoc[]
+
+include::partial$snip_finding-postgresql-pod.adoc[]
+
+include::partial$snip_dumping-che-database.adoc[]
+
+include::partial$snip_droping-che-database.adoc[]
+
+include::partial$snip_creating-che-database.adoc[]
+
+. Migrate {prod-short} users:
++
+[subs="+quotes,+attributes"]
+----
+DUMP="che.sql"
+ALL_USERS={prod-id}-users.txt
```

Comment:
@AndrienkoAleksandr I tried to verify the procedure, but there are no instructions for generating `eclipse-che-users.txt`
Running the script from step 4 just prints smth.like: 
```
[INFO] Find Che user: d5cce9c5-9586-415a-bc5e-ff4b03b187ee and corresponding OpenShift user: 0caf37d3-0ae4-40ff-8b1a-23f7b6049b0d
[INFO] Done.
```

Reply:
sorry, my bad the file is generated on step 4 - https://github.com/eclipse/che-docs/pull/2159/files#diff-77c9e22f032894ec21ce593ccfa6ec9c032e1847c7da18431442f931b803b026R30